hs_composite_resource/models.py

```diff
@@ -65,17 +66,76 @@ def can_be_published(self):
             # url file cannot be published
             if 'url' in lf.extra_data:
                 return False
+            # check for model instance linked to model program
+            if lf.is_model_instance:
+                mi_aggr = lf
+                mp_aggr = mi_aggr.metadata.executed_by
+                if mp_aggr is not None:
+                    if mp_aggr.resource.short_id != mi_aggr.resource.short_id:
+                        # model instance aggregation and model program aggregations are not in the same resource
+                        # the other resource containing the linked model program must have been published
+                        # before this resource can be published
+                        if not mp_aggr.resource.raccess.published:
+                            return False
+                    else:
+                        # model program aggregation and the linked model instance aggregations are in the same resource
+                        continue
+                model_program_url = mi_aggr.metadata.executed_by_url
+                if model_program_url:
```

Comment:
Is this block necessary?  The above logic checks if the resource is published... if the resource is published, you should get the doi url.  If a resource i published and doesn't have  a doi url, we've got a resource in a dirty state and if anything, we should be logging a warning or something.  I don't think we should even be checking it here.  If we have published resources without a doi url, then we should be checking it elsewhere and working to patch that bug.